husarion_ugv_manager/include/husarion_ugv_manager/plugins/shutdown_host.hpp

```diff
@@ -120,152 +142,89 @@ class ShutdownHost
     }
   }
 
+  void Halt() override { command_handler_->Halt(); }
+
+  std::string GetIp() const override { return ip_; }
+  std::string GetError() const override { return failure_reason_; }
+  std::string GetOutput() const override { return command_handler_->GetOutput(); }
+  ShutdownHostState GetState() const override { return state_; }
+
+protected:
   bool IsAvailable() const
   {
     return system(("ping -c 1 -w 1 " + ip_ + " > /dev/null").c_str()) == 0;
   }
 
-  void CloseConnection()
+  /**
+   * @brief Get the current time since epoch in seconds.
+   *
+   * @return std::int64_t The current time since epoch in seconds.
+   */
+  std::int64_t GetTimeSinceEpoch()
   {
-    if (ssh_channel_is_closed(channel_)) {
-      return;
-    }
-
-    ssh_channel_send_eof(channel_);
-    ssh_channel_close(channel_);
-    ssh_channel_free(channel_);
-    ssh_disconnect(session_);
-    ssh_free(session_);
+    return std::chrono::duration_cast<std::chrono::seconds>(
+             std::chrono::system_clock::now().time_since_epoch())
+      .count();
   }
```

Comment:
The function looking like this suggests that it is important that we are running time_since_epoch, but I don't quite understand why it would matter, we can't just use
```suggestion
  std::int64_t GetTimeSinceEpoch()
  {
    return std::chrono::duration_cast<std::chrono::seconds>(
             std::chrono::system_clock::now()).count();
  }
```
And if we can, I suggest adding this function to *_utils.

Reply:
system_clock::now() returns a `time_point`, and we need `duration` to use count. Also, shutdown http server expects to pass timestamp of the request in format of "time since epoch in seconds". In that context this method is quite specific for calling http server. Where in utils would you suggest to place this?